### PR TITLE
Enhancement/Lightspin AWS IAM vulnerability

### LINKED ISSUE #1246: AWS: Add support "Lightspin AWS IAM Vulnerability"
References:
* https://blog.lightspin.io/aws-iam-groups-authorization-bypass
* https://github.com/lightspin-tech/red-shadow

Important bit:
```
At first glance it looks like it will provide very good protection since deny overrides any allows and if access is denied for a group than it also denied for the group members. But in fact, in AWS - it doesn’t. The deny implies ONLY for Group object actions and all other User object actions are still allowed.
```

As per the code, all we need is a finding that checks policies for a specific `Resource` (https://github.com/lightspin-tech/red-shadow/blob/main/red-shadow.py#L5) and `Actions` (https://github.com/lightspin-tech/red-shadow/blob/main/red-shadow.py#L6).

### PR DESCRIPTION
**Pull request by lowSoA**
# Description

Added a new finding for "Lightspin AWS IAM vulnerability". Further details can be found in the [original writeup](https://blog.lightspin.io/aws-iam-groups-authorization-bypass).

Also fixed a typo in [ScoutSuite/providers/aws/rules/findings/iam-root-account-with-active-keys.json](https://github.com/nccgroup/ScoutSuite/commit/a7caed46091df176c1668927354552750435358b#diff-93e85871f5518035a8d8e734dad6fd47e059879808afbf70e7e2956c219fff3d) which generated parsing errors.

Fixes #1246

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (optional)
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: ScoutSuite/providers/aws/rules/findings/iam-lightspin-user-action-denied-for-group.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
+{
+    "description": "Policy with Denied User Actions for Group Objects (Lightspin Vulnerability)",
```
Comment (j4v) [inline ScoutSuite/providers/aws/rules/findings/iam-lightspin-user-action-denied-for-group.json:2]:
```suggestion
    "description": "Policy with Denied User Actions for Group Objects",
```

**Review by j4v — APPROVED** [review]

**Review by alessandrogonzalez — APPROVED** [review]
LGTM

File: ScoutSuite/core/conditions.py [hunk 2]
```diff
@@ -181,6 +181,18 @@ def pass_condition(b, test, a):
             if re.match(c, b):
                 result = True
                 break
+    elif test == 'matchInList':
```
Comment (alessandrogonzalez) [inline ScoutSuite/core/conditions.py:184]: Could you please add some tests to https://github.com/nccgroup/ScoutSuite/blob/master/tests/test_utils_conditions.py so codecov does not complain?

**Review by alessandrogonzalez — APPROVED** [review]
LGTM

**Pull request merged**